supervisor/host/network.py

```diff
@@ -74,11 +69,17 @@ def dns_servers(self) -> List[str]:
 
     async def check_connectivity(self):
         """Check the internet connection."""
+        if HostFeature.NETWORK not in self.sys_host.supported_features:
+            self._connectivity = None
+            return
         try:
             state = await self.sys_dbus.network.check_connectivity()
-            self._connectivity = ConnectivityState(state[0])
+
+            # ConnectionState 4 == FULL (has internet)
+            # https://developer.gnome.org/NetworkManager/stable/nm-dbus-types.html#NMConnectivityState
+            self._connectivity = state[0] == 4
         except (DBusError, IndexError):
```

Comment:
I think the index is not needed if the call is successful
```suggestion
        except DBusError:
```

Reply:
You are probably right, we _should_ always get a code here.